lib/_http_client.js

```diff
@@ -182,7 +182,7 @@ function ClientRequest(options, cb) {
                       'client');
     }
     self._storeHeader(self.method + ' ' + self.path + ' HTTP/1.1\r\n',
-                      self._headers);
+                      self._headersStore);
```

Comment:
Ok so it might be worth sticking to properties for now.

As far as adding the benchmark, it could be useful. Although I'd probably put it in misc/ instead of es/ since it doesn't primarily deal with es6+ alternatives (which es/ seems to contain).